node.go

```diff
@@ -41,15 +42,15 @@ type immutableProtoNode struct {
 // this mutable protonode implementation is needed to support go-unixfs,
 // the only library that implements both read and write for UnixFS v1.
 type ProtoNode struct {
+	mu sync.RWMutex
```

Comment:
does this protect the entire ProtoNode? if not, document what it does protect

Reply:
It should protect all of it, yes... aside from the misses below I am about to fix - what prompted the question?

Reply:
I wondered if any of the fields were effectively read-only after construction, so the mutex wouldn't protect them.